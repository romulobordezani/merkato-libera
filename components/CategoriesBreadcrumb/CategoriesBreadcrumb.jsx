import React from 'react';
import PropTypes from 'prop-types';

import styles from './CategoriesBreadcrumb.module.scss';

const CategoriesBreadcrumb = ({ item = null }) => {
  return (
    <div className={styles['categories-breadcrumb']}>
      <div className={styles['categories-breadcrumb_container']}>
        {item &&
          item.categories.map(category => (
            <span className={styles['categories-breadcrumb_item']}>{category}</span>
          ))}
        {!item && <span className={styles['categories-breadcrumb_item']}>&nbsp;</span>}
      </div>
    </div>
  );
};

CategoriesBreadcrumb.propTypes = {
  item: PropTypes.arrayOf({
    categories: PropTypes.shape({
      id: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired
    })
  })
};

export default CategoriesBreadcrumb;
