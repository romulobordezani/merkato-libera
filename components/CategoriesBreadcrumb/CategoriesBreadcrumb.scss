@import "../_SassLib/breakpoints";
@import "../_SassLib/colors";

.categories-breadcrumb {
  font-size: 14px;
  background: transparent;
  display: grid;
  grid-template-rows: auto;
  $mobilePadding: 0.2%;
  grid-template-columns: $mobilePadding 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr $mobilePadding;
  grid-template-areas: ". BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid .";
  column-gap: 1%;
  margin: 0 2%;
  height: 100%;

  @include breakpoint($small) {
    margin-bottom: 16px;
  }

  &_container {
    grid-area: BreadGrid;
  }

  &_item{
    display: inline-block;
    margin: 0 15px;
    padding: 16px 0;
    color: $grey;

    @include breakpoint($small) {
      display: none;
    }

    &:before {
      content: " > ";
      position: relative;
      left: -13px;
      top: 1px;
      width: 10px;
      height: 10px;
      color: grey;
    }

    &:first-child {
      &:before {
        content: "";
        display: none;
      }
    }

    &:last-child {
      font-weight: 600;
    }
  }
}


.grid-container {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-template-rows: 1fr;
  grid-template-areas: ". BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid BreadGrid .";
}

.BreadGrid { grid-area: BreadGrid; }
