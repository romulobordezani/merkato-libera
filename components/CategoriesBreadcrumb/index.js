import React from 'react';

import styles from './CategoriesBreadcrumb.scss';

const CategoriesBreadcrumb = () => <div className={styles['categories-breadcrumb']} />;

export default CategoriesBreadcrumb;
