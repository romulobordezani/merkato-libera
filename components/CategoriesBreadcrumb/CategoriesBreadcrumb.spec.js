import React from 'react';
import renderer from 'react-test-renderer';

import CategoriesBreadcrumb from './CategoriesBreadcrumb';

describe('CategoriesBreadcrumb', () => {
  it('Snapshot renders default', () => {
    expect(renderer.create(<CategoriesBreadcrumb />)).toMatchSnapshot();
  });
});
