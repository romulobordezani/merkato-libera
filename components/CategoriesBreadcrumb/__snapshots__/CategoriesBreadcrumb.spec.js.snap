// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesBreadcrumb Snapshot renders default 1`] = `
<div>
  <div>
    <span>
      Â 
    </span>
  </div>
</div>
`;
